doc/sphinx/Pacemaker_Explained/rules.rst

```diff
@@ -554,63 +524,226 @@ A small sample of how time-based expressions can be used:
-   :widths: 1 3
-
-   +---------------+-----------------------------------------------------------+
-   | Attribute     | Description                                               |
-   +===============+===========================================================+
-   | id            | .. index::                                                |
-   |               |    pair: id; rsc_expression                               |
-   |               |                                                           |
-   |               | A unique name for this element (required)                 |
-   +---------------+-----------------------------------------------------------+
-   | class         | .. index::                                                |
-   |               |    pair: class; rsc_expression                            |
-   |               |                                                           |
-   |               | The standard name to be matched against resource agents   |
-   +---------------+-----------------------------------------------------------+
-   | provider      | .. index::                                                |
-   |               |    pair: provider; rsc_expression                         |
-   |               |                                                           |
-   |               | If given, the vendor to be matched against resource       |
-   |               | agents (only relevant when ``class`` is ``ocf``)          |
-   +---------------+-----------------------------------------------------------+
-   | type          | .. index::                                                |
-   |               |    pair: type; rsc_expression                             |
-   |               |                                                           |
-   |               | The name of the resource agent to be matched              |
-   +---------------+-----------------------------------------------------------+
-
-Example Resource-Based Expressions
-__________________________________
+Resource Type Expressions
+#########################
+
+The ``rsc_expression`` element *(since 2.0.5)* configures a rule condition
+based on the agent used for a resource. It is allowed in rules in a
+``meta_attributes`` element within a ``rsc_defaults`` or ``op_defaults``
+element.
+
+.. list-table:: **Attributes of a rsc_expression Element**
+   :class: longtable
+   :widths: 1 1 1 4
+   :header-rows: 1
+
+   * - Name
+     - Type
+     - Default
+     - Description
+   * - .. _rsc_expression_id:
+
+       .. index::
+          pair: rsc_expression; id
+
+       id
+     - :ref:`id <id>`
+     - 
+     - A unique name for this element (required)
+   * - .. _rsc_expression_class:
+
+       .. index::
+          pair: rsc_expression; class
+
+       class
+     - :ref:`text <text>`
+     - 
+     - If this is set, the expression is satisfied only if the resource's agent
+       standard matches this value
```

Comment:
I tend to use "standard" mainly because it's more natural language and "class" makes many people think of the programming construct. As a C++ reserved word, class as a variable name causes problems with some header checks and static analysis, but it's easy enough to use rsc_class instead. Documentation has always used both interchangeably. I don't have a strong opinion, and I'd be fine with picking one.

Reply:
I'm good with preferring "standard" going forward. I've always used `typename` instead of `class` in C++ templates for the same reason.